### PR TITLE
Refactor work package index styling (flexible box based solution)

### PR DESCRIPTION
**Pull request by myabc**
This PR has a broad scope and also includes:
- some refactoring of legacy styles (`main.css`)
- commit from #1605

https://www.openproject.org/work_packages/14811
https://www.openproject.org/work_packages/13498
https://www.openproject.org/work_packages/13597

as collateral:
https://www.openproject.org/work_packages/11935

### CODE REVIEW

**Comment by myabc** [comment]
By no means perfect, but nevertheless ready for a preliminary review.

**Comment by ulferts** [comment]
When reducing the window size by reducing the browser vertically, the menu becomes scrollable. It is desired for it to scroll vertically, but a scroll bar is also present horizontally.

File: app/assets/stylesheets/layout/_split_view.sass [hunk 1]
```diff
@@ -1,313 +0,0 @@
...
-  .split-view .work-packages--details, .work-packages--split-view .work-packages--details
-    display: none
-  li[feature-flag=detailsView]
-    display: none
-@media only screen and (min-width: 1281px)
-  .split-view .work-packages--details, .work-packages--split-view .work-packages--details
-    width: 480px
-@media only screen and (min-width: 1440px)
-  .split-view .work-packages--details, .work-packages--split-view .work-packages--details
-    width: 580px
```
Comment (ulferts) [inline app/assets/stylesheets/layout/_split_view.sass:313]: @myabc, you (accidentally?) removed the media query specifying different width for different resolutions. Now the detail pane is rather narrow on my screen.

**Comment by myabc** [comment]
I am not able to reproduce the cuke failure locally (`cucumber features/work_packages/index_pagination.feature`)

File: app/assets/stylesheets/layout/_split_view.sass (hunk 1 above)
Comment (myabc) [inline app/assets/stylesheets/layout/_split_view.sass:313]: @ulferts those media query rules were moved to `work_packages.sass` (line 76).

File: app/assets/stylesheets/content/_work_packages.sass [hunk 2]
```diff
@@ -27,24 +27,102 @@
...
+    +flex-basis(100%)
 
 .work-packages--split-view
-  @extend .split-view
+  +display(flex)
+  width: 100%
 
 .work-packages--list
-  @extend %panel
+  +flex-basis(100%)
```
Comment (ulferts) [inline app/assets/stylesheets/content/_work_packages.sass]: @myabc this causes the problems with the detail pane being too narrow. If you change this to 

<pre>
flex-grow: 1;
</pre>

it works for me (Chrome). Works when the detail pane is displayed and also when it isn't. Media query is also working. Plus I believe that this is what you want to express. 
See: http://css-tricks.com/snippets/css/a-guide-to-flexbox/

Comment (myabc) [inline app/assets/stylesheets/content/_work_packages.sass]: @ulferts thanks. I realised last night that `flex-basis: 100%` was causing the details pane width to be ignored. I'll give this a try.

**Comment by myabc** [comment]
Tested in latest Chrome, Firefox, Safari, [Firefox 24.0](https://saucelabs.com/bugs/daa2a93511b84d288daf127d28d458c0), IE 10, 11.

Also tested (and broken) in [Firefox 17](https://saucelabs.com/bugs/30e7590271e64cde9a5f3f2ca13fd017).

**Comment by ulferts** [comment]
@myabc this looks really good :+1: 

The problem I spotted has to do with the flash messages, as they will push down everything. As long as the flash message is there, the bottom of the page (pagination & edit-button) will be pushed into the void.

**Comment by ulferts** [comment]
Can't reproduce the failing cuke either.

**Comment by ulferts** [comment]
@myabc another thing I noticed is that we probably need a min height for all of this. When I reduce the height very much, the elements begin to be placed over one another.

**Comment by myabc** [comment]
@hschink would you be able to do me a favour and see if you can reproduce
the  cuke failure locally?

**Comment by manwithtwowatches** [comment]
@myabc Can't scroll down on the create work package view anymore

**Comment by ulferts** [comment]
@myabc opening the context menu for a row at the bottom of the page, will open the context menu as desired. But as the context menu is opened downwards it will reach over the end of the page leading to scroll bars for the whole page. The cheap solution would be to add more space to the bottom to handle those cases. But this will destroy the layout. So I guess the menu will have to open upwards. 

image

**Comment by ulferts** [comment]
@myabc there is less space between the tab selection and the heading on the global work packages page compared to the space on the project specific work packages page. 
image

**Comment by myabc** [comment]
@ulferts 

> there is less space between the tab selection and the heading on the global work packages page compared to the space on the project specific work packages page. 

should be addressed in 51bbdd5

**Comment by myabc** [comment]
@ulferts 

> So I guess the menu will have to open upwards.

I think that solution would make the most sense, but it would be a fix that would take more than a few minutes. I'd rather deal with it as a separate WP/PR.

**Comment by ulferts** [comment]
@myabc 

> I think that solution would make the most sense, but it would be a fix that would take more than a few minutes. I'd rather deal with it as a separate WP/PR.

Fine with that. 

I spotted another tiny mistake (sorry for getting on your nerves with those) and that also has to do with a menu:

image

When the last column in the work packages table has less width than the context menu, opening the context menu will increase the width of the table (or it's container). The solution could be to place the menu differently in that case. 

Whether you want to address this now or in a separate PR is up to you. I believe there are more pressing bugs.

**Comment by myabc** [comment]
@ulferts 

> Whether you want to address this now or in a separate PR is up to you. I believe there are more pressing bugs.

I'd prefer to make context menu adjustments in a separate PR.

**Comment by ulferts** [comment]
@myabc space and missing scrolling fixed. :+1:

**Comment by ulferts** [comment]
:clap:

**Pull request merged**